UnchainedLauncher.Core/Services/Processes/Chivalry/LaunchPreparers/Chivalry2ModsInstaller.cs

```diff
@@ -0,0 +1,181 @@
+                        releasesList
+                        .Map(_modRegistry.GetModRelease)
+                        .Map(async t => await t)
+                    )
+                    ).SplitEithers();
+
+            var failuresCount = failures.Fold(0, (c, e) => {
+                _logger.Error($"Failed to get registry metadata: {e.Message}");
+                return c + 1;
+            });
+
+            if (failuresCount != 0) {
+                return None;
+            }
+
+            // delete any paks not mentioned in this launch
+            // TODO: do something more clever than just deleting. This weirdness ultimately comes
+            // from the fact that launches share a pak dir, and can affect each other.
+            // This stuff is only really relevant to servers, and there's probably not much
+            // we can do without adding some seriously in-depth hooks in the plugin
+            // that allows doing some kind of per-process pak dir isolation
+            var modFileNames = metas.Map(m => m.PakFileName);
+            foreach (string file in Directory.EnumerateFiles(FilePaths.PakDir)
+                     .Filter(f => f.Contains(".pak"))
+                     .Filter(f => !f.Contains("pakchunk0-WindowsNoEditor.pak"))
+                     .Filter(f => !modFileNames.Any(n => f.Contains(n)))) {
+                try {
+                    File.Delete(file);
+                }
+                catch (Exception e) {
+                    _logger.Error($"Failed to delete '{file}'", e);
+                }
+            }
+
+            // tracks versions of installed paks
+            var installedFiles = new LastInstalledPakVersionMeta(FilePaths.LastInstalledPakVersionList);
+            var downloadsMap = new Map<ReleaseCoordinates, Release>()
+                .AddRange(releases.Zip(metas))
+                .Filter(release => {
+                    var alreadyExists = File.Exists(Path.Join(FilePaths.PakDir, release.PakFileName));
+                    var isDifferentVersion = installedFiles.UpdateVersion(ReleaseCoordinates.FromRelease(release));
+                    if (isDifferentVersion || !alreadyExists) {
+                        return true;
+                    }
+
+                    _logger.Info($"Not overwriting already installed and versioned mod {release.PakFileName}");
+                    return false;
+
+                });
+
+            var cts = new CancellationTokenSource(6000);
+
+            // TODO: display download progress as a popup or something
+            var downloads = await Task.WhenAll(
+                downloadsMap.Pairs.Map(
+                    p => _modRegistry.DownloadPak(p.Item1, Path.Join(FilePaths.PakDir, p.Item2.PakFileName))
+                ).Map(
+                    e => e.Map(
+                        writer => writer.WriteAsync(None, cts.Token)
+                        )
+                    ).Map(e => e.MatchAsync(
+                        r => r.Match(
```

Comment:
I'm surprised the formatter doesn't catch this, but generally the `p =>` part should go on the line above the body. It'll help reduce line length and make it more consistent with the other pieces here